prov/verbs/src/fi_verbs.c

```diff
@@ -224,16 +224,15 @@ static int fi_ibv_sockaddr_len(struct sockaddr *addr)
 	}
 }
 
-static int fi_ibv_check_fabric_attr(const struct fi_fabric_attr *attr)
+static int fi_ibv_check_fabric_attr(const struct fi_fabric_attr *attr,
+		const struct fi_info *info)
 {
-	if (attr->name && !(!strcmp(attr->name, VERBS_ANY_FABRIC) ||
-	    !strncmp(attr->name, VERBS_IB_PREFIX, strlen(VERBS_IB_PREFIX)) ||
-	    !strcmp(attr->name, VERBS_IWARP_FABRIC))) {
+	if (attr->name && strcmp(attr->name, info->fabric_attr->name)) {
 		FI_INFO(&fi_ibv_prov, FI_LOG_CORE, "Unknown fabric name\n");
 		return -FI_ENODATA;
 	}
 
-	if (attr->prov_version > VERBS_PROV_VERS) {
+	if (attr->prov_version > info->fabric_attr->prov_version) {
```

Comment:
Because this is checking the value from a parameter, the check looks wrong.  The check must be against the version defined by the verbs code, and not through some parameter, which could have come from the app.

I *think* the confusion is that the 'info' parameter is really referring to the global variable verbs_info.  If that's the case, then I would reference verbs_info directly, as a global to indicate the intent.